ee/clickhouse/views/test/test_clickhouse_experiments.py

```diff
@@ -200,24 +202,89 @@ def test_experiment_flow_with_event_results(self):
+        self.assertEqual("test", result[1][0]["breakdown_value"][0])
 
         self.assertEqual(result[1][1]["name"], "$pageleave")
-        self.assertEqual(result[1][1]["count"], 2)
-        self.assertEqual("control", result[1][1]["breakdown_value"][0])
+        self.assertEqual(result[1][1]["count"], 1)
+        self.assertEqual("test", result[1][1]["breakdown_value"][0])
+
+        # Variant with test: Beta(2, 3) and control: Beta(3, 1) distribution
+        # The variant has very low probability of being better.
+        self.assertAlmostEqual(response_data["probability"], 0.357, places=3)
+
+
+class ClickhouseTestTrendExperimentResults(ClickhouseTestMixin, LicensedTestMixin, APIBaseTest):
+    @snapshot_clickhouse_queries
+    def test_experiment_flow_with_event_results(self):
+        journeys_for(
+            {
+                "person1": [
+                    {"event": "$pageview", "timestamp": "2020-01-02", "properties": {"$feature/a-b-test": "test"},},
+                ],
+                "person2": [
+                    {"event": "$pageview", "timestamp": "2020-01-03", "properties": {"$feature/a-b-test": "control"}},
+                ],
+                "person3": [
+                    {"event": "$pageview", "timestamp": "2020-01-04", "properties": {"$feature/a-b-test": "control"}},
+                ],
+                # doesn't have feature set
+                "person_out_of_control": [{"event": "$pageview", "timestamp": "2020-01-03",},],
+                "person_out_of_end_date": [
+                    {"event": "$pageview", "timestamp": "2020-08-03", "properties": {"$feature/a-b-test": "control"}},
+                ],
+            },
+            self.team,
+        )
+
+        ff_key = "a-b-test"
+        # generates the FF which should result in the above events^
+        response = self.client.post(
+            f"/api/projects/{self.team.id}/experiments/",
+            {
+                "name": "Test Experiment",
+                "description": "",
+                "start_date": "2020-01-01T00:00",
+                "end_date": "2020-01-06T00:00",
+                "feature_flag_key": ff_key,
+                "parameters": None,
+                "filters": {
+                    "insight": "trends",
+                    "events": [{"order": 0, "id": "$pageview"}],
+                    "display": "ActionsLineGraphCumulative",
+                    "properties": [
+                        {"key": "$geoip_country_name", "type": "person", "value": ["france"], "operator": "exact"}
+                        # properties superceded by FF breakdown
+                    ],
+                },
+            },
+        )
+
+        id = response.json()["id"]
+
+        response = self.client.get(f"/api/projects/{self.team.id}/experiments/{id}/results")
+        self.assertEqual(200, response.status_code)
+
+        response_data = response.json()
+        result = sorted(response_data["insight"], key=lambda x: x["breakdown_value"])
```

Comment:
Nit: What does the sorted array look like? You could perhaps use `assertCountEquals` instead .

Reply:
I found this annoying, because then I'll have to put in all the other elements of the dictionary as well.

All I want to confirm here is: when the breakdown value is `control` (or `test`), then the count values are appropriate.

The problem with not sorting is that the ordering is not deterministic, which makes things flakey. After sorting, I know the first element will always be control

Reply:
You can also use ANY from unittest.mock module for these values :thinking: But leaving it up to you.